// @enable3d/common/dist/extendedMesh.js
/**
 * @author       Yannick Deubel (https://github.com/yandeu)
 * @copyright    Copyright (c) 2021 Yannick Deubel; Project Url: https://github.com/enable3d/enable3d
 * @license      {@link https://github.com/enable3d/enable3d/blob/master/LICENSE|LGPL-3.0}
 */
import { Mesh, Vector3 } from 'three';
export class ExtendedMesh extends Mesh {
    ptr;
    isExtendedMesh = true;
    isGroup = false;
    vector3 = new Vector3();
    shape;
    name;
    body;
    hasBody = false;
    // convex object breaking
    fragmentDepth = 0;
    breakable = false;
    fractureImpulse = 1;
    constructor(geometry, material) {
        super(geometry, material);
        this.name = `object-${this.id}`;
    }
    /** Returns all values relative to the world. */
    get world() {
        return {
            theta: this.worldTheta,
            phi: this.worldPhi
        };
    }
    /** Get the theta relative to the world. */
    get worldTheta() {
        this.getWorldDirection(this.vector3);
        return Math.atan2(this.vector3.x, this.vector3.z);
    }
    /** Get the phi relative to the world. */
    get worldPhi() {
        this.getWorldDirection(this.vector3);
        return Math.acos(this.vector3.y);
    }
}
